stablehlo/dialect/TypeInference.cpp

```diff
@@ -2544,7 +2544,15 @@ LogicalResult inferSelectOp(
 
 LogicalResult inferSelectAndScatterOp(
     Value operand, SmallVectorImpl<Type>& inferredReturnTypes) {
-  inferredReturnTypes.push_back(operand.getType());
+  auto operandRankedType = operand.getType().dyn_cast<RankedTensorType>();
+  if (!operandRankedType) {
```

Comment:
nit: could you reverse the 2 branches of if-else, so that "If(!xxx)" can be avoid.